lib/OpenQA/WebAPI/Controller/API/V1/Asset.pm

```diff
@@ -176,15 +175,19 @@ sub delete {
         $cond{type} = $args{type};
     }
     else {
-        return;
+        return undef;
     }
 
-    my $asset = $self->app->schema->resultset("Assets")->search(\%cond, \%attrs);
-    return unless $asset;
-    my $rs = $asset->delete_all;
+    my $asset = $self->schema->resultset("Assets")->search(\%cond);
+    return $self->render(json => {error => 'Asset not found. The asset might have been removed.'}, status => 404)
```

Comment:
Maybe "The asset might have *already* been removed and only the cached view has not bee updated yet.".

Reply:
fixed it